Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarWebServiceTest.cs

```diff
@@ -940,26 +941,54 @@ public async Task GetProperties_Old_Forbidden()
         [TestMethod]
         public void GetProperties_63plus_Forbidden()
         {
-            const string serverUrl = "http://localhost";
-            const string projectKey = "my-project";
-
             var downloaderMock = new Mock<IDownloader>();
             downloaderMock
-                .Setup(x => x.Download(new Uri($"{serverUrl}/api/server/version"), false))
+                .Setup(x => x.Download(new Uri($"{ServerUrl}/api/server/version"), false))
                 .Returns(Task.FromResult("6.3.0.0"));
 
             downloaderMock
-                .Setup(x => x.TryDownloadIfExists(new Uri($"{serverUrl}/api/settings/values?component={projectKey}"), true))
+                .Setup(x => x.TryDownloadIfExists(new Uri($"{ServerUrl}/api/settings/values?component={ProjectKey}"), true))
                 .Throws(new HttpRequestException("Forbidden"));
 
-            var service = new SonarWebService(downloaderMock.Object, serverUrl, logger);
+            var service = new SonarWebService(downloaderMock.Object, ServerUrl, logger);
 
-            Action action = () => _ = service.GetProperties(projectKey, null).Result;
+            Action action = () => _ = service.GetProperties(ProjectKey, null).Result;
             action.Should().Throw<HttpRequestException>();
 
             logger.Errors.Should().HaveCount(1);
         }
 
+        [TestMethod]
+        public void DownloadCache_NullArguments()
+        {
+            ws.Invoking(x => x.DownloadCache(null, "branch")).Should().ThrowAsync<ArgumentNullException>();
+            ws.Invoking(x => x.DownloadCache("key", null)).Should().ThrowAsync<ArgumentNullException>();
```

Comment:
over-nitpicking: Using `key123` will make this test a bit more readable
```suggestion
            ws.Invoking(x => x.DownloadCache("key123", null)).Should().ThrowAsync<ArgumentNullException>();
```